resources/idf-python-wheels_diagram_source.txt

```diff
@@ -0,0 +1,79 @@
+@startuml
+
+:build-wheels-platforms.yml;
+start
+
+:Install OS dependencies (os_dependencies directory);
+
+:Build wheels for IDF (build_wheels.py)
+
+- minimum IDF-supported Python version;
+start
+
+:Fetch all ESP-IDF branches;
+
+:Get used ESP-IDF branches (specified from GitHub variables);
+
+:Get constraints files for used ESP-IDF branches;
+
+:                      Assemble requirements
+- create requirements.txt
+- transform yaml to requirement from exclude_list.yaml
+- exclude defined requirements from requirements.txt
+- create additional requirements from include_list.yaml;
+
+:         Build additional requirements
+- from include_list.yaml
+- print requirements to be built wheels for
+- print statistics;
+if (1 or more wheel(s) build failed) then (yes)
+:Raise Error;
+stop
+endif
+
+:                 Build requirements
+- from requirements.txt after exclude
+- print requirements to be built wheels for
+- print statistics;
+if (1 or more wheel(s) build failed) then (yes)
+:Raise Error;
+stop
+endif
+
+:                Get Python version dependent wheels
```

Comment:
Can we describe this in a more detail please. This one wasn't clear, at least to me, on the first look.  Maybe shortly explain what we mean by python dependent wheels. Something like

Python dependent wheels are wheels which depend  on the cpython [ABI](https://docs.python.org/3/c-api/stable.html). These are checked based on the [wheel filename format](https://peps.python.org/pep-0491/#file-format) where the `abi tag` is check for `cp`. Such wheels need to be build also on all supported python version, not only on the minimum python version supported by `ESP-IDF`.

Or something like this.

Other than this small note I think it looks great. Thank you for doing this.

Reply:
Thank you for review, @fhrbata. Sure, this is a good suggestion I have stopped thinking about that since I have seen it so many times, it became a term for me.

I will add a note.

Reply:
Your explanation is pretty good, so I have used it, if you are OK with this @fhrbata?

Thank you!

[README.md preview ](https://github.com/espressif/idf-python-wheels/blob/docs/diagram/README.md)